src/classes/PMT_PaymentCreator.cls

```diff
@@ -319,29 +319,33 @@ public class PMT_PaymentCreator {
                     }
                     
                     // we may need to add payments if the open opp amount has been changed from $0
-                    else if (thisOpp.Amount != 0 && thisOpp.Amount != null && (oldOpp.Amount == 0 || oldOpp.Amount == null ) 
-                        && !thisOpp.isClosed && !isNoPaymentForOpp(thisOpp)){
-                        npe01__oppPayment__c op = new npe01__oppPayment__c(
-                            npe01__opportunity__c = thisOpp.id,
-                            npe01__Paid__c = false,
+                    else if (
```

Comment:
Overall, the code looks. In general, I usually check for null before checking for a value.